eq-author-api/db/index.js

```diff
@@ -1,3 +1,37 @@
-var config = require("../config/knexfile.js");
+const pinoMiddleware = require("express-pino-logger");
+const createLogger = require("../utils/createLogger");
+const pino = pinoMiddleware();
+const logger = createLogger(pino.logger);
 
-module.exports = require("knex")(config);
+const config = require("../config/knexfile.js");
+
+const getConfig = async function(secretId) {
+  const result = {
+    ...config
+  };
+  logger.log("Getting DB Credentials");
+  if (secretId && secretId !== "") {
+    logger.log("Getting DB Credentials from AWS Secrets Manager");
+    const AWS = require("aws-sdk");
+    const client = new AWS.SecretsManager({ region: "eu-west-1" });
+    try {
+      const secret = await client
+        .getSecretValue({ SecretId: secretId })
+        .promise();
+      const dbCredentials = JSON.parse(secret.SecretString);
+      result.connection = {
+        host: dbCredentials.host,
+        port: dbCredentials.port,
+        user: dbCredentials.username,
+        password: dbCredentials.password,
+        database: dbCredentials.dbname
```

Comment:
It looks like we are just naming dbname to datbase. Can just do that and spread the rest?

Reply:
How do I do that?

Reply:
```js
const { dbName, ...otherProps } = dbCredentials;

result.connection = {
  database: dbName,
  ...otherProps
}
```